### PR TITLE
Compatible API for KBC

### PR DESCRIPTION
**Pull request by Xynnn007**
This PR aims to resolve the compatible problem for KBCes, including the following:
- Defined a common AnnotationPacket format for all KBCes.
- Brought decryption logic out of each KBC module
- Added documents and specs
- Added new feature openssl to support hardware acceleration for some HW

Warning: This pr will make current CI (image decryption) using sample-kbc, eaa-kbc not work. Thus we need to make new test images after the next Pr which will update the key provider module

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,11 +97,11 @@ List of supported KBC modules:
 
 | KBC module name    | README                                                              | KBS protocol | Maintainer                |
 | ------------------ | ------------------------------------------------------------------- | ------------ | ------------------------- |
-| sample_kbc         | Null                                                                | Null         | Attestation Agent Authors |
 | offline_fs_kbc     | [Offline file system KBC](src/kbc_modules/offline_fs_kbc/README.md) | Null         | IBM                       |
 | eaa_kbc            | [EAA KBC](src/kbc_modules/eaa_kbc/README.md)                        | EAA protocol | Alibaba Cloud             |
 | offline_sev_kbc    | [Offline SEV KBC](src/kbc_modules/offline_sev_kbc/README.md)        | Null         | IBM                       |
 | online_sev_kbc     | [Online SEV KBC](src/kbc_modules/online_sev_kbc/README.md)          | simple-kbs   | IBM                       |
+| cc_kbc             | [CC KBC](src/kbc_modules/cc_kbc/README.md)                          | generic-kbs  | CoCo Community            |
```
Comment (jialez0) [inline README.md:104]: `generic-kbs` ->  `[CoCo KBS protocol](https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md)`

File: src/crypto/aes256ctr.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
```
Comment (fitzthum) [inline src/crypto/aes256ctr.rs:4]: We have a `src/common` directory already. Would that be a good place to put this crypto stuff?

**Review by fitzthum — COMMENTED** [review]
LGTM. A few small comments. 

This PR unifies the encryption, but it doesn't use the new resource URI, which will be a bit more complicated. Looks like it will not change the behavior of most KBCs at all.

**Review by fitzthum — APPROVED** [review]

**Comment by Xynnn007** [comment]
@fitzthum Thanks for your comments. Yes, resource URI and other behaviors will be updated in the following PRs.

**Review by jialez0 — APPROVED** [review]

**Pull request merged**